concent_api/core/views.py

```diff
@@ -35,6 +36,8 @@
 @transaction.non_atomic_requests(using='control')
 def send(_request: HttpRequest, client_message: Message, client_public_key: bytes) -> Union[Message, HttpResponse]:
     assert isinstance(client_public_key, bytes) or client_public_key is None
+    if not are_all_stored_messages_compatible_with_protocol_version(client_message, client_public_key):
+        return concents.ServiceRefused(reason=concents.ServiceRefused.REASON.InvalidRequest)
```

Comment:
I think the same comment about new reason might be applied here.

Reply:
Similar answer as above :) It is moved to message_handlers

Reply:
Make it a separate function instead of mixing responsibilities.

Reply:
I created function `check_protocol_versions_and_update_subtasks_from_incoming_message_if_timed_out`. Reason is to ask database only one about subtask and check everything we need using one db query instead of twice asking database at the same thing. Is it ok?